src/lib/auth.ts

```diff
@@ -28,24 +48,36 @@ const createAuth = () => {
     set({ user: null, known: true })
   }
 
+  // This is a helper function to get the auth provider for login.
+  // It will not be avialble in the export.
   function providerFor(name: string) {
     switch (name) {
       case 'google':   return new GoogleAuthProvider()
       default:         throw 'unknown provider ' + name
     }
   }
 
+  // This function will be available as `auth.signInWith(providerName)`
+  // Calling it will trigger an authStateChange event, causing
+  // the listener set above to set the auth store with the new value.
   async function signInWith(name: string) {
     const auth = getAuth(app)
     const provider = providerFor(name)
     await signInWithRedirect(auth, provider)
   }
 
+  // This function will be avaialble as `auth.signOut()`
+  // Calling it will trigger an authStateChange event, causing
+  // the listener set above to set the auth store with the new value.
   async function signOut() {
     const auth = getAuth(app)
     await _signOut(auth)
   }
 
+  // Here is the store that is returned as the export.
+  // Now we can `import { auth } from '$lib/auth'`,
+  // and in our components, we can call `$auth` to subscribe to the value of the auth store, 
+  // or we can call `auth.signInWith(providerName)` or `auth.signOut()` without subscribing.
```

Comment:
Maybe just needs a reference to where it's used as an example of consuming the state and calling the store's methods:
https://github.com/CaptainCodeman/sveltekit-example/blob/master/src/lib/Auth.svelte